Aptoide/aptoide-client-v8
# Feature/an 1498 timelinev2 codestyle

What does this PR do?

   This pull-request brings a full rehaul to the Apps Timeline: Apps Timeline v2.
  
   Most relevant changes:
-  Social stats info in cards ( number of likes/comments ) is now displayed in a different card section (single bar) and only shows up if there is at least 1 like or comment.
-  Cards layout changed according to zeplin : [Timeline v2](https://app.zeplin.io/project/58a5dde3079da73c80cc892e/screen/58de9269bbe91e3e1434a874).
-  Similar cards were deleted, they are now parsed as Recommendation cards in TimelineItem.java.
-  Custom fonts were deleted from the Timeline.
-  All Cards views were migrated to Aptoide Guideline styles.
-  A new row was added to the social cards containing the Latest comment.
-  A new card was added: Popular App (among your friends).
-  Like/Comment on Non-social cards are now silent (does not show share preview dialog).
-  Social comment text (in the social info stats row) goes to the comments list.
-  Clicking comment balloon goes to the comments list with comment dialog popping up.
-  Every app card (Recommendation/Social Rec/Update/Popular/SocialInstall) is now shared with storeID.
-  New cards: Aggregated (article/video/install/store latest), this cards aggregated shares from multiple people (share same content) and have sub-cards (which I call minimal cards) that are the original shared cards model and have social actions.
- Aggregated cards have show more (when more than 2 people shared the same card) and should expand to show more minimal cards.
- Share Preview dialogs were adjusted to the layout changes and added to the aggregated cards.

Where should the reviewer start?

- [ ] Test social info (social_info_bar_timeline.xml) if visibility flow is right.
- [ ] Test if Similar cards are being parsed as Recommendation cards, if they show up at all.
- [ ] Test latest comment on social cards.
- [ ] Test Popular App card (if it doesn't show up, ask Diogo Pires).
- [ ] Test like & comment on non-social cards (should share+like/comment on the shared card).
- [ ] Test if commenting flow (click on balloon, click on comments number).
- [ ] Test if app cards shares have a storeId field.
- [ ] Test aggregated cards (UI).
- [ ] Test aggregated cards (Headers info).
- [ ] Test aggregated cards (Minimal cards).
- [ ] Test aggregated cards (actions).
- [ ] Test every share preview dialog.
- [ ] Test share app in AppView and Installed apps (updates tab).

    

How should this be manually tested?

   Open Aptoide v8 > Apps Timeline > Check if what comes from the getUserTimeline request matches what is being shown on cards (specially Social ones).
   Make interactions with cards (create a share of every card & reshare of every social card - TimelineItem.java for all the card types)
   For tests in social cards the Timeline by userId can be used (ana2016   store for example).

Any background context you want to provide?

N/A
What are the relevant tickets?

Tickets related to this pull-request: [Popular App](https://aptoide.atlassian.net/browse/AN-1493), [Timeline revision](https://aptoide.atlassian.net/browse/AN-1498), [Removing custom Fonts](https://aptoide.atlassian.net/browse/AN-1508), [Silent Like/comment](https://aptoide.atlassian.net/browse/AN-1522), [Share with storeId](https://aptoide.atlassian.net/browse/AN-1523),[Aggregated Cards](https://aptoide.atlassian.net/browse/AN-1536)

Screenshots (if appropriate)

N/A
Questions:

   Is there a blog post? N/A
   Does this add new dependencies which need to be added? No
